### PR TITLE
EVG-8288: do not allow configuration of commit queue patches

### PR DESCRIPTION
**Pull request by tgrander**
https://jira.mongodb.org/browse/EVG-8288

### Summary
Screen Shot 2020-07-07 at 2 37 38 PM

- 'Reconfigure' link in dropdown menu on patch card (user patches page) and on version page is disabled for patches that are on the commit queue. 
- Users are redirected away from configure page and to the version page if patch is on commit queue

### Testing
- E2E tests
- manually tested both features listed above in staging

### CODE REVIEW

File: src/pages/ConfigurePatch.tsx [hunk 1]
```diff
@@ -10,17 +10,31 @@ import { ConfigurePatchCore } from "pages/configurePatch/ConfigurePatchCore";
...
     variables: { id },
   });
+
   usePageTitle(`Configure Patch`);
 
+  const router = useHistory();
+
+  // redirect to version page if patch is on commit queue
+  useEffect(() => {
+    const commitQueuePosition = data?.patch?.commitQueuePosition ?? null;
```
Comment (john-m-liu) [inline src/pages/ConfigurePatch.tsx:31]: I believe this check fails if the patch has already started. I would check if data.patch.alias = "__commit_queue" to determine if it's a commit queue patch. Same with the other spot

File: src/pages/ConfigurePatch.tsx [hunk 2]
```diff
@@ -10,17 +10,31 @@ import { ConfigurePatchCore } from "pages/configurePatch/ConfigurePatchCore";
...
+
+    if (commitQueuePosition !== null) {
+      router.push(getVersionRoute(id));
```
Comment (SupaJoon) [inline src/pages/ConfigurePatch.tsx:34]: we should replace instead of push to keep the browser history clean

File: src/pages/Patch.tsx [hunk 3]
```diff
@@ -42,16 +46,23 @@ const PatchCore: React.FC = () => {
...
 
   const patch = get(data, "patch");
   const status = get(patch, "status");
   const description = get(patch, "description");
   const activated = get(patch, "activated");
+
+  const commitQueuePosition = patch?.commitQueuePosition ?? null;
+  const isPatchOnCommitQueue = commitQueuePosition !== null;
+
   if (activated === false) {
```
Comment (SupaJoon) [inline src/pages/Patch.tsx:60]: Does this condition need to be adjusted so we don't redirect if `isPatchOnCommitQueue` is true?

File: src/pages/ConfigurePatch.tsx (hunk 1 above)
Comment (tgrander) [reply src/pages/ConfigurePatch.tsx:31]: great suggestion, thanks!

File: src/pages/ConfigurePatch.tsx (hunk 2 above)
Comment (tgrander) [reply src/pages/ConfigurePatch.tsx:34]: will do

File: src/pages/Patch.tsx (hunk 3 above)
Comment (tgrander) [reply src/pages/Patch.tsx:60]: yeah!

**Pull request merged**